src/core/thread/mle.hpp

```diff
@@ -666,6 +666,17 @@ class Mle : public InstanceLocator
     {
         return mAllNetworkBackboneRouters.GetAddress();
     }
+
+    /**
+     * This method returns a reference to the All Domain Backbone Routers Multicast Address.
+     *
+     * @returns A reference to the All Domain Backbone Routers Multicast Address.
+     *
+     */
+    const Ip6::Address &GetAllDomainBackboneRoutersAddress(void) const
+    {
+        return mAllDomainBackboneRouters.GetAddress();
```

Comment:
`mAllDomainBackboneRouters` is not properly set up when BBR service is not enabled or when there is no domain prefix in network. 
I think it's not good to return a wrong multicast address. Should we return error instead in those cases?

Reply:
Note that the `AllNetworkBBRs`/`AllDomainBBRs` are only used for BTMF:
- `AllNetworkBBRs` is for `b/bmr` when BBR service is enabled and accepts `n/mr`.
- `AllDomainBBRs` is for `b/bq` when BBR service is enabled, domain prefix is available and accepts `n/dr`.

I would prefer to keep the code simple as is considering the use scenarios in core so far, and open to do possible enhancements if required in the future. 

Thoughts?

Reply:
> `AllDomainBBRs` is for `b/bq` when BBR service is enabled, domain prefix is available and accepts `n/dr`.

Guarantee these conditions by `OT_ASSERT`?

Reply:
Like said above, bbr may not process n/dr if no domain prefix, it won't use AllDomainBBRs.
Besides, like existing [Mle::GetMeshLocal16()](https://github.com/openthread/openthread/blob/master/src/core/thread/mle.hpp#L744), it simply return the address without assert the attachment status.
I prefer to keep the code simple as is when we know clearly the address is there when it is used.